scripts/linters/pylint_extensions.py

```diff
@@ -1654,99 +1654,88 @@ class SingleLineCommentChecker(checkers.BaseChecker):
-                continue
-
-            # Check if comment contains any excluded phrase.
-            word_is_present_in_excluded_phrases = any(
-                word in line for word in EXCLUDED_PHRASES)
+        for (token_type, _, (line_num, _), _, line) in tokens:
+            if token_type == tokenize.COMMENT:
+                line = line.strip()
 
-            # Comments may include a lowercase character at beginning
-            # or may not use a punctuation at end if it contains a
-            # excluded phrase e.g. "# coding: utf-8".
-            if word_is_present_in_excluded_phrases:
-                continue
+                if not line.startswith('#'):
+                    continue
 
-            if not next_line.startswith(b'#'):
-                # Comments must end with the proper punctuation.
-                last_char_is_invalid = line[-1] not in (
-                    ALLOWED_TERMINATING_PUNCTUATIONS)
-                if last_char_is_invalid:
+                # Comments must start with a space.
+                if (re.search(br'^#[^\s].*$', line) and not
+                        line.startswith(b'#!')):
+                    space_at_beginning_of_comment = False
                     self.add_message(
-                        'invalid-punctuation-used', line=line_num + 1)
+                        'no-space-at-beginning', line=line_num)
 
-            # Comments must start with a space.
-            if re.search(br'^#[^\s].*$', line) and not line.startswith(b'#!'):
-                space_at_beginning_of_comment = False
-                self.add_message(
-                    'no-space-at-beginning', line=line_num + 1)
-
-            # Comments may include a lowercase character at the beginning
-            # only if they start with version info or a data type or.
-            # a variable name e.g. "# next_line is of string type."
-            # or "# v2 version does not have ExplorationStatsModel."
-            # or "# int. The file size, in bytes.".
-            if len(line) > 1 and space_at_beginning_of_comment:
-
-                # Check if variable name is used.
-                underscore_is_present = '_' in line.split()[1]
-                if underscore_is_present:
-                    continue
+                if line_num - previous_line_num > 1:
+                    split_line = line.split()
+                    split_prev_line = previous_line.split()
+
+                    # Check if comment contains any excluded phrase.
+                    excluded_phrase_is_present_at_beginning = any(
+                        word in split_line[1] for word in
+                        EXCLUDED_PHRASES)
+
+                    if len(previous_line) > 1:
+                        excluded_phrase_is_present_at_end = any(
+                            word in split_prev_line[-1] for word in
+                            EXCLUDED_PHRASES)
+                        excluded_phrase_at_beginning_of_prev_line = any(
```

Comment:
Why we need this?

Reply:
to check for `pylint` at the beginning of comment, we do not use a period in that case.